cfme/cloud/flavor.py

```diff
@@ -97,25 +98,106 @@ def is_displayed(self):
 
     breadcrumb = BreadCrumb()
     title = Text('//div[@id="main-content"]//h3')
-    entities = View.nested(BaseNonInteractiveEntitiesView)
+    # In case, to check reset flash message
+    flash = FlashMessages(".//[@id='flash_text_div']")
+    category = BootstrapSelect("tag_cat")
+    tag = BootstrapSelect("tag_add")
     save = Button('Save')
     reset = Button('Reset')
     cancel = Button('Cancel')
 
 
+class FlavorCollection(Navigatable):
+    def instantiate(self, name, provider):
+        return Flavor(
+            name,
+            provider,
+            collection=self
+        )
+
+    def add_tag(self, tag, *flavors):
+        """Add given tag to flavor collections
+
+        Args:
+            tag: tuple describing tags
+            flavors: list of :py:class:`cfme.cloud.flavor.Flavor` objects
+        """
+        flavors = list(flavors)
+        checked_flavors = []
+        view = navigate_to(FlavorCollection, 'All')
+        # double check we're in List View
+        view.toolbar.view_selector.select('List View')
+        if not view.entities.table.is_displayed:
+            raise ValueError('No flavor found')
+        for row in view.entities.table:
+            for flavor in flavors:
+                if flavor == row.name.text:
+                    checked_flavors.append(flavor)
+                    row[0].check()
+                    break
+            if set(flavors) == set(checked_flavors):
+                break
+        if set(flavors) != set(checked_flavors):
+            raise ValueError('Some flavors were not found in the UI')
+        view.toolbar.policy.item_select('Edit Tags')
+        mixins.add_tag(tag, navigate=False)
+
+    def remove_tag(self, tag, *flavors):
+        """Remove given tag from flavor collections
+
+        Args:
+            tag: tuple describing tags
+            flavors: list of :py:class:`cfme.cloud.flavor.FlavorCollection` objects
+        """
+        flavors = list(flavors)
```

Comment:
**Required**
There is only one line difference between this and `add_tag`, consider pulling all of this common code out into a method that both of them can call instead. Otherwise we have to maintain both code blocks when any bugs or changes arise.

Reply:
Makes sense, changing.